framework/contracts/account/manager/src/commands.rs

```diff
@@ -210,6 +227,37 @@ pub(crate) fn install_modules_internal(
     ))
 }
 
+fn find_nonce(
```

Comment:
Won't it be more efficient/predictable to get the nonce for each module individually instead of getting the minimal nonce for all modules?

Reply:
Also this fn generates the salt that you end up re-generating after this fn. Maybe it can return the salt(s) itself?

Reply:
Yeah I mean it's a subtle difference...

Maybe we shouldn't even allow for re-installing?

Reply:
Because I could see some really cool applications with 100% predictable module addresses. 
As soon as we add this nonce it becomes <100% ...

Reply:
> Yeah I mean it's a subtle difference...
> 
> Maybe we shouldn't even allow for re-installing?

implemented that!